crates/trie/sparse-parallel/src/trie.rs

```diff
@@ -841,6 +980,198 @@ impl SparseSubtrie {
         current_level == child_level
     }
 
+    /// Updates or inserts a leaf node at the specified key path with the provided RLP-encoded
+    /// value.
+    ///
+    /// This method updates the internal prefix set and, if the leaf did not previously exist,
```

Comment:
> updates the internal prefix set

Where is this happening? The only prefix set updates I see are in the `ParallelSparseTrie::update_leaf` call.

Actually I think maybe this comment might just be out of date, it also mentions returning the split branch

Reply:
yeah I just copy pasted the comment from update_leaf, will update